### PR TITLE
Document function get_memory in autogpt.memory.vector

### PR DESCRIPTION
**Pull request by sagarishere**
Document function get_memory in ./scripts/memory/init.py

### Background
def get_memory was not documented. Now it is.

Documentation helps greatly some developers, when they can move the pointer over a function and read exactly what it does.

### Changes
Just Documentation

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

File: scripts/memory/__init__.py
Comment (joshmarsden1994) [inline scripts/memory/__init__.py]: [nit] - some formatting suggestions:

```suggestion
    """
    Return a memory object based on the specified memory backend in the configuration.

    :param cfg: A configuration object that contains information about the memory
        backend to be used and other relevant parameters.
    :param init: A boolean parameter that indicates whether to initialize the memory
        or not. If set to True, the memory will be cleared. Defaults to False (optional).

    :return: An instance of a memory object based on the configuration provided. The
        type of memory object returned depends on the value of the `memory_backend`
        attribute in the configuration. If `memory_backend` is set to "pinecone", a
        `PineconeMemory` object is returned. If it is set to "redis", a `RedisMemory`
        object is returned. By default, a `LocalCache` object is returned.
    """
```

**Review by joshmarsden1994 — APPROVED** [review]

File: scripts/memory/__init__.py
Comment (sagarishere) [reply scripts/memory/__init__.py]: Hi, there is an issue with spacing. I know that many functions before have spaces in the documentation. Now if we put spaces, it doesn't pass the automated CD/CI pipeline checks. I did try to do it like you have suggested and it could not pass the tests/checks of CD/CI. 

Hope this clarifies.

Comment (sagarishere) [reply scripts/memory/__init__.py]: and thank you for the suggestion, it is a really good suggestion. 

Some changes need to be made in CD-CI pipeline before we can put spaces in doc strings.

Comment (joshmarsden1994) [reply scripts/memory/__init__.py]: 💡 I see, thanks for the heads up on that. I will take a look at the formatting and linting in the pipeline separately to see if there's anything that can be done.

**Comment by sjnt1** [comment]
Sorry for the noob question but is my redis stack server / cotainer  working properly when it shows this all the time : 

2023-04-14 19:42:21 42:C 14 Apr 2023 17:42:21.019 * DB saved on disk
2023-04-14 19:42:21 42:C 14 Apr 2023 17:42:21.019 * RDB: 0 MB of memory used by copy-on-write

Seems like its not storing any data?

**Comment by sagarishere** [comment]
Hi. Replying to the Redis question. It works. Nothing to worry about. This is overhead memory indicator. This is an ideal number

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Review by dschonholtz — DISMISSED** [review]
This is a small, useful, atomic change that adds value. 
Approved.
If it is not passing lint, make sure it does before merging. 
I am not worried about the nit on spacing. If you can make that work while it continues to pass linting stuff that's great. If you can't get it to play nice with the linter with the suggested changes, just merge it where it passes the lint check.

Thanks for the contribution!

**Pull request merged**
